mmengine/dataset/base_dataset.py

```diff
@@ -531,15 +530,15 @@ def _join_prefix(self):
         """
         # Automatically join annotation file path with `self.root` if
         # `self.ann_file` is not an absolute path.
-        if not is_abs(self.ann_file) and self.ann_file:
-            self.ann_file = osp.join(self.data_root, self.ann_file)
+        if self.ann_file and not is_abs(self.ann_file):
+            self.ann_file = osp.join(self.data_root or '', self.ann_file)
         # Automatically join data directory with `self.root` if path value in
         # `self.data_prefix` is not an absolute path.
         for data_key, prefix in self.data_prefix.items():
             if isinstance(prefix, str):
                 if not is_abs(prefix):
                     self.data_prefix[data_key] = osp.join(
-                        self.data_root, prefix)
+                        self.data_root or '', prefix)
```

Comment:
```python
                if not is_abs(prefix) and self.data_root:
                    self.data_prefix[data_key] = osp.join(
                        self.data_root, prefix)
```

Reply:
Even if `data_root=''`, this code should also be executed